src/python/pants/backend/jvm/tasks/classpath_util.py

```diff
@@ -154,7 +155,11 @@ def classpath_entries_contents(cls, classpath_entries):
         # Walk the jar namelist.
         with open_zip(entry, mode='r') as jar:
           for name in jar.namelist():
-            yield name
+            # Zip entry names can come in any encoding and in practice we find some jars that have
+            # utf-8 encoded entry names, some not.  As a result we cannot simply decode in all cases
+            # and need to do this to_bytes(...).decode('utf-8') dance to stay safe across all entry
+            # name flavors and under all supported pythons.
+            yield to_bytes(name).decode('utf-8')
```

Comment:
IIRC, the right way to do the `to_bytes` conversion is to use `six`: https://pythonhosted.org/six/#six.binary_type , which would let you drop the pex dep.